### PR TITLE
publish nightlies to bintray

### PR DESCRIPTION
**Pull request by eed3si9n**
This PR is on top of #1981.
Here's a SNAPSHOT published to my personal repo test-test-test: https://bintray.com/eed3si9n/test-test-test/sbt/0.13.9-20150420-123655/view#files

/review @jsuereth

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -11,7 +11,12 @@ import Sxr.sxr
 // but can be shared across the multi projects.
 def buildLevelSettings: Seq[Setting[_]] = Seq(
   organization in ThisBuild := "org.scala-sbt",
-  version in ThisBuild := "0.13.9-SNAPSHOT"
+  version in ThisBuild := "0.13.9-SNAPSHOT",
+  // bintrayOrganization in ThisBuild := None,
+  // bintrayRepository in ThisBuild := "test-test-test",
+  bintrayOrganization in ThisBuild := Some("sbt"),
+  bintrayRepository in ThisBuild := s"ivy-${(publishStatus in ThisBuild).value}",
```
Comment (jsuereth) [inline build.sbt:18]: nice

File: project/Release.scala [hunk 2]
```diff
@@ -1,62 +1,20 @@
...
+import _root_.bintray.BintrayPlugin.autoImport._
+import _root_.bintray.InternalBintrayKeys._
 
-object Release extends Build {
-  lazy val remoteBase = SettingKey[String]("remote-base")
-  lazy val remoteID = SettingKey[String]("remote-id")
+object Release {
   lazy val launcherRemotePath = SettingKey[String]("launcher-remote-path")
   lazy val deployLauncher = TaskKey[Unit]("deploy-launcher", "Upload the launcher to its traditional location for compatibility with existing scripts.")
-  lazy val checkCredentials = TaskKey[Unit]("checkCredentials", "Checks to ensure credentials for this user exists.")
```
Comment (jsuereth) [inline project/Release.scala:12]: bintray-sbt already has checkCredentials right?  Would be sad to loose this kind of check before releasing.

**Comment by jsuereth** [comment]
LGTM.  We'll have to also edit the jenkins config to have credentials...

File: project/Release.scala (hunk 2 above)
Comment (eed3si9n) [inline project/Release.scala:12]: Yea. If you didn't have Bintray auth file, warnings would show up on shell.

File: project/StatusPlugin.scala [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
   override def buildSettings: Seq[Setting[_]] = Seq(
-    isSnapshot <<= version(v => v.contains("-") && snapshotQualifier(v)),
-    publishStatus <<= isSnapshot { snap => if (snap) "snapshots" else "releases" },
+    isSnapshot := {
+      val SnapshotQualifier = """(.+)(-.*SNAPSHOT)(.*)""".r
+      val v = version.value
+      v match {
+        case SnapshotQualifier(_, _, _) => true
+        case _                          => false
+      }
```
Comment (dwijnand) [inline project/StatusPlugin.scala:21]: good old:

```
version.value matches "(.+)(-.*SNAPSHOT)(.*)"
```

?

Comment (eed3si9n) [inline project/StatusPlugin.scala:21]: Yea. I aped the original implementation but likely `version.value contains "-SNAPSHOT"` would work just as well.

Comment (jsuereth) [inline project/StatusPlugin.scala:21]: pattern matching LIKE IT AIN'T NO THANG.

**Comment by eed3si9n** [comment]
Under https://ci.typesafe.com/configure, "sbt Linux JDK6" AMI now points ami-904544f8 (used to be ami-a28483ca).

**Pull request merged**